### ISSUE TITLE
Update NumberInput accessibility

### ISSUE THREAD

**Issue by nakayan5** [yamada-ui/yamada-ui#2014]
### Description

This issue aims to address accessibility deficiencies in our NumberInput component as per W3C guidelines.
The component currently does not meet the accessibility standards outlined in the W3C documentation.

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

- [x] If the spinbutton has a visible label, it is referenced by [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) on the spinbutton element. Otherwise, the spinbutton element has a label provided by [aria-label](https://w3c.github.io/aria/#aria-label).

### Problem Statement/Justification

improve accessibility

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

**Comment by nakayan5** [comment]
related https://github.com/yamada-ui/yamada-ui/issues/1506

**Comment by cidkumagai** [comment]
I'll try it.
